src/service/standard-actions-impl.js

```diff
@@ -125,6 +131,21 @@ export class StandardActions {
    * @param {!./action-impl.ActionInvocation} invocation
    * @private
    */
+  handleAmpNavigateTo_(invocation) {
+    if (!invocation.satisfiesTrust(ActionTrust.HIGH)) {
+      return;
+    }
+    const url = invocation.args['url'];
+    const expandedUrl = this.urlReplacements_.expandUrlSync(url);
+    const node = invocation.target;
+    const win = (node.ownerDocument || node).defaultView;
+    win.location = expandedUrl;
```

Comment:
For nested iframes? I haven't heard this particular FR and it can be accomplished by binding to `iframe[src]` with amp-bind. We could extend this to `myIframeId.navigateTo(url=)` if requested.